extension/src/pipeline/quickPick.ts

```diff
@@ -61,109 +68,160 @@ const pickTemplateAndFields = async (
-  )
+  return getMetricInfoFromArr(maybeFieldsObjArr)
+}
 
-const getFieldsFromDataFiles = (
-  dataArr: { data: UnknownValue; file: string }[]
-) => {
-  const keys: {
-    [file: string]: string[]
-  } = {}
-  let keysAmount = 0
+const dvcPlotGuidelinesQuestion =
+  'Does the file contain data and follow the DVC plot guidelines for [JSON/YAML](https://dvc.org/doc/command-reference/plots/show#example-hierarchical-data) or [CSV/TSV](https://dvc.org/doc/command-reference/plots/show#example-tabular-data) files?'
+
+const validateSingleFileData = ({
+  file,
+  data
+}: {
+  data: UnknownValue
+  file: string
+}) => {
+  const { fields = [] } = getMetricInfoFromValue(data) || {}
+
+  if (fields.length < 2) {
+    void Toast.showError(
+      `${file} does not contain enough keys (columns) to generate a plot. ${dvcPlotGuidelinesQuestion}`
+    )
+    return
+  }
 
-  for (const { file, data } of dataArr) {
-    const fields = getFieldsFromValue(data)
+  return [{ fields, file }]
+}
 
-    if (fields.length === 0) {
-      void showNotEnoughKeysToast()
+const validateMultiFilesData = (
+  filesData: { data: UnknownValue; file: string }[]
+) => {
+  const filesArrLength: Set<number> = new Set()
+  const keys: FileFields = []
+
+  for (const { file, data } of filesData) {
+    const metricInfo = getMetricInfoFromValue(data)
+    if (!metricInfo) {
+      void Toast.showError(
+        `${file} does not contain any keys (columns) to generate a plot. ${dvcPlotGuidelinesQuestion}`
+      )
       return
     }
 
-    keysAmount += fields.length
-
-    if (!keys[file]) {
-      keys[file] = []
-    }
-    keys[file].push(...fields)
+    const { arrLength, fields } = metricInfo
+    keys.push({ fields, file })
+    filesArrLength.add(arrLength)
   }
 
-  if (keysAmount < 2) {
-    void showNotEnoughKeysToast()
+  if (filesArrLength.size > 1) {
+    void Toast.showError(
+      'All files must have the same array (list) length. See [examples](https://dvc.org/doc/command-reference/plots/show#sourcing-x-and-y-from-different-files) of multiple files being used in DVC plots.'
+    )
+
     return
   }
 
   return keys
 }
 
+const validateFilesData = async (cwd: string, files: string[]) => {
+  const filesData = (await loadDataFiles(files)) as {
+    data: UnknownValue
+    file: string
+  }[]
+  const relativeFilesData = []
+
+  for (const { file, data } of filesData) {
+    const relativeFile = relative(cwd, file)
+    if (!data) {
+      void Toast.showError(
+        `Failed to parse ${relativeFile}. ${dvcPlotGuidelinesQuestion}`
+      )
+      return
+    }
+
+    relativeFilesData.push({ data, file: relativeFile })
+  }
+
+  return relativeFilesData.length === 1
+    ? validateSingleFileData(relativeFilesData[0])
+    : validateMultiFilesData(relativeFilesData)
+}
+
 export const pickPlotConfiguration = async (
```

Comment:
Summary of error handling:

* Check if file(s) have the same extension
*  Check if file(s) have data
*  If Single File
*  * Check if file has at least two fields to pick from
* If Multiple Files
* * Check if each file has at least one field to pick from (toast is thrown at first invalid file found)
* * Check if all files have the same array length